app/scripts/lib/sentry.js

```diff
@@ -0,0 +1,133 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+define([
+  'underscore',
+  'raven',
+  'lib/url'
+], function (_, Raven, Url) {
+
+  var ALLOWED_QUERY_PARAMETERS = [
+    'client_id',
+    'keys',
+    'redirect_uri',
+    'scope',
+    'style',
+    'verification_redirect',
+    'campaign',
+    'customizeSync',
+    'entrypoint',
+    'context',
+    'migration',
+    'service',
+    'setting',
+    'automatedBrowser'
+  ];
+
+  /**
+   * function that gets called before data gets sent to error metrics
+   *
+   * @param {Object} data
+   *  Error object data
+   * @returns {Object} data
+   *  Modified error object data
+   * @private
+   */
+  function beforeSend(data) {
+    if (data && data.request && data.request.url) {
+      data.request.url = cleanUpQueryParam(data.request.url);
+    }
+
+    return data;
+  }
+
+  /**
+   * Sanitize and clear extra query parameter information
```

Comment:
This function doesn't clear any extra query params, rather it overwrites the value of a query parameter if it's not allowed.